packages/flame/test/components/route_test.dart

```diff
@@ -474,6 +475,51 @@ void main() {
         );
       },
     );
+    testWithFlameGame('Route with loading', (game) async {
+      final loadingComponent = PositionComponent(size: Vector2.all(100));
+      final pageComponent = _HeavyComponent()..size = Vector2.all(100);
+      final router = RouterComponent(
+        initialRoute: 'start',
+        routes: {
+          'start': Route(
+            Component.new,
+          ),
+          'new': Route(
+            () => pageComponent,
+            loadingBuilder: () => loadingComponent,
+          ),
+        },
+      )..addToParent(game);
+      await game.ready();
+
+      router.pushNamed('new');
+
+      // I can't perform this following checking, probably because the way it 
+      // was implemented, in assumption that line 512 actually logically works.
+      // If line 512, is doing nothing then I don't think, I can perform this 
+      // following check either, that is to check state when loading was 
+      // mounted before it was removed.
+
+      // expect(
+      //   loadingComponent.isMounted,
+      //   isTrue,
+      // );
+      // expect(
+      //   pageComponent.isMounted,
+      //   isFalse,
+      // );
```

Comment:
I don't think I follow why this doesn't work, what are you getting here currently?

Reply:
I'm sorry , False is being expected for 503, as for 507 it does pass

Reply:
Why is false expected for 503? Since the route is added it should be showing the `loadingComponent` right?

Reply:
I just found, I made an error in the checking at line 514 is not accurate in this commit, it should be expecting True, but I checked there for False and it passed

That means my loading component was never mounted even before and after the main component was mounted? 
Maybe, that mean my implementation may have some error ?

Reply:
I'm guessing that due to how `Future.delayed` works in tests the `loadingComponent` was added and removed in the same tick. You could pass in a future to the `_HeavyComponent` that you manually complete after you have checked that the `loadingComponent` is mounted.

Reply:
Did you have any more time to look at this @PistonShot1?

Reply:
Sorry for the delays got caught up with another project, but I am trying to implement the approach you mentioned , to pass a future that I can manually complete, I am not exactly sure how to implement this . I am assuming with `Completer` or sort ?

Reply:
Yeah, `Completer` should work

Reply:
the test for checking if the `loadingComponent` was mounted and remove, both doesn't pass (at line 499 and at line 509), as I have commented, it is some how implying that the loadingPage was never mounted. I am not sure why, but the changes I added at `packages/flame/lib/src/components/route.dart` (line 146-151), supposed have mounted and removed after awaiting `_page!.loaded;`